### PR TITLE
fix: use primary theme color for entity star icon

### PR DESCRIPTION
**Pull request by callebstrom**
### Background
TLDR; we should use primary theme color for the entity star (favorite) icon. See https://github.com/backstage/backstage/issues/26385.  

### Changes
- Remove all occurrences of `#f3ba37` as color for `Star`

### Screenshots 
image

image

image

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [x] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Comment by benjdlambert** [comment]
Hmm, wondering if we should add an item for this in the theme to be honest, It's a little strange to use the primary colour I'm feeling. Unless there's a better fit for it in the pallette? Some accent colour or something? :pray:

**Comment by callebstrom** [comment]
@benjdlambert we seem to be using the primary color as default for the other icons in similar contexts, such as `SettingsIcon` in the `UserListPicker` in catalog. 

Or is your point more that the favourite icon should be treated differently _because_ we used to have a separate color for this?
